File: easyserver/src/main/java/org/ecjtu/easyserver/server/util/ImageUtil.java
package org.ecjtu.easyserver.server.util;

import android.graphics.Bitmap;
import android.media.MediaMetadataRetriever;
import android.media.ThumbnailUtils;
import android.os.Build;
import android.provider.MediaStore;

import java.io.FileOutputStream;
import java.lang.Exception;
import java.lang.IllegalArgumentException;
import java.lang.RuntimeException;
import java.util.HashMap;

/**
 * Created by KerriGan on 2017/7/11.
 */

public class ImageUtil {

    public static boolean saveBitmap(Bitmap bitmap, String path, Bitmap.CompressFormat format, int quality) {
        try {
            bitmap.compress(format, quality,new FileOutputStream(path));
            return true;
        } catch (Exception e){
            return false;
        }
    }

    /**
     * 使用android内部native库,不需要转utf-8编码
     *
     * @param url httpUrl
     */
    public static Bitmap createVideoThumbnail(String url, int width, int height) {
        Bitmap bitmap = null;
        MediaMetadataRetriever retriever =new MediaMetadataRetriever();
        int kind = MediaStore.Video.Thumbnails.MINI_KIND;

        try {
            if (Build.VERSION.SDK_INT >= 14) {
                retriever.setDataSource(url,new HashMap < String, String > ());
            } else {
                retriever.setDataSource(url);
            }

            bitmap = retriever.getFrameAtTime();
        } catch (IllegalArgumentException ex){
            // Assume this is a corrupt video File
        } catch(RuntimeException ex){
            // Assume this is a corrupt video File.
        } finally{
            try {
                retriever.release();
            } catch (RuntimeException ex){
                // Ignore failures while cleaning up.
            }

        }
        if (kind == android.provider.MediaStore.Images.Thumbnails.MICRO_KIND && bitmap != null) {
            bitmap = ThumbnailUtils.extractThumbnail(bitmap, width, height,
                    ThumbnailUtils.OPTIONS_RECYCLE_INPUT);
        }
        return bitmap;
    }
}

File: app/src/main/java/com/ecjtu/sharebox/util/image/ImageUtil.kt
package com.ecjtu.sharebox.util.image

import android.graphics.Bitmap
import android.graphics.Canvas
import android.media.MediaMetadataRetriever
import android.media.ThumbnailUtils
import android.os.Build
import android.provider.MediaStore
import java.io.FileOutputStream
import java.lang.Exception
import java.util.HashMap
import android.graphics.PixelFormat
import android.graphics.drawable.Drawable



/**
 * Created by KerriGan on 2017/7/11.
 */

object ImageUtil{
    @JvmStatic
    fun saveBitmap(bitmap:Bitmap, path:String, format: Bitmap.CompressFormat, quality:Int):Boolean{
        try {
            bitmap.compress(format,quality,FileOutputStream(path))
            return true
        }catch (e:Exception){
            return false
        }
    }

    /**
     * 使用android内部native库,不需要转utf-8编码
     * @param url httpUrl
     */
    @JvmStatic
    fun createVideoThumbnail(url: String, width: Int, height: Int): Bitmap? {
        var bitmap: Bitmap? = null
        val retriever = MediaMetadataRetriever()
        val kind = MediaStore.Video.Thumbnails.MINI_KIND

        try {
            if (Build.VERSION.SDK_INT >= 14) {
                retriever.setDataSource(url, HashMap<String, String>())
            } else {
                retriever.setDataSource(url)
            }

            bitmap = retriever.frameAtTime
        } catch (ex: IllegalArgumentException) {
            // Assume this is a corrupt video File
        } catch (ex: RuntimeException) {
            // Assume this is a corrupt video File.
        } finally {
            try {
                retriever.release()
            } catch (ex: RuntimeException) {
                // Ignore failures while cleaning up.
            }

        }
        if (kind == android.provider.MediaStore.Images.Thumbnails.MICRO_KIND && bitmap != null) {
            bitmap = ThumbnailUtils.extractThumbnail(bitmap, width, height,
                    ThumbnailUtils.OPTIONS_RECYCLE_INPUT)
        }
        return bitmap
    }

    @JvmStatic
    fun drawable2Bitmap(drawable: Drawable): Bitmap {
        val bitmap = Bitmap.createBitmap(
                drawable.intrinsicWidth,
                drawable.intrinsicHeight,
                if (drawable.opacity != PixelFormat.OPAQUE)
                    Bitmap.Config.ARGB_8888
                else
                    Bitmap.Config.RGB_565)

        val canvas = Canvas(bitmap)
        drawable.setBounds(0, 0, drawable.intrinsicWidth, drawable.intrinsicHeight)
        drawable.draw(canvas)
        return bitmap
    }
}